xarray/core/formatting.py

```diff
@@ -47,22 +47,11 @@ def wrap_indent(text, start='', length=None):
     return start + indent.join(x for x in text.splitlines())
 
 
-def ensure_valid_repr(string):
-    """Ensure that the given value is valid for the result of __repr__.
-
-    On Python 2, this means we need to convert unicode to bytes. We won't need
-    this function once we drop Python 2.7 support.
-    """
-    if PY2 and isinstance(string, unicode_type):
-        string = string.encode('utf-8')
-    return string
-
-
 class ReprMixin(object):
```

Comment:
If you're feeling ambitious, you could drop this mixin class entirely, after replacing all `__unicode__` methods with `__repr__`.